ext/openssl/xp_ssl.c

```diff
@@ -110,6 +115,21 @@
 #define PHP_X509_NAME_ENTRY_TO_UTF8(ne, i, out) \
 	ASN1_STRING_to_UTF8(&out, X509_NAME_ENTRY_get_data(X509_NAME_get_entry(ne, i)))
 
+/* Used for IPv6 Address peer verification */
+#define EXPAND_IPV6_ADDRESS(_str, _bytes) \
+	do { \
+		snprintf(_str, 39, "%X:%X:%X:%X:%X:%X:%X:%X", \
```

Comment:
```suggestion
		snprintf(_str, 40, "%X:%X:%X:%X:%X:%X:%X:%X", \
```
This should be the buffer size. With 39 it might discard the last character here...